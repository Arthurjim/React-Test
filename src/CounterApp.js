import React, { useState } from "react";
import PropTypes from "prop-types";

// snipett rafcp
function CounterApp({ value = 10 }) {
   const [counter, setCounter] = useState(value);
   // aumenta en 1
   const handleAdd = (e) => {
      setCounter(counter + 1);
      // sertCounter((c)=>c+1)
   };

   return (
      <>
         <h1>CounterApp</h1>
         <h2>{counter}</h2>

         <button onClick={handleAdd}>+1</button>
         <button onClick={()=>{setCounter(value)}}>Reset</button>
         <button onClick={()=>setCounter(counter -1)}>-1</button>
      </>
   );
}

CounterApp.defaultProps = {
   // value: 100,
};
CounterApp.propTypes = {
   value: PropTypes.number,
};

export default CounterApp;
