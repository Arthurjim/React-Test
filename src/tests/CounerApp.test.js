import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Prueba en <CounterApp />", () => {
    
    // let wrapper
    //se hace as√≠ para establecer el valor por defecto del  wrapper y poner tener autocompletoado
    let wrapper =shallow(<CounterApp />);
    //se ejecuta antes de cada test
    beforeEach(()=>{
         wrapper = shallow(<CounterApp />);
    })

   test("Debe ser igual el Counter app", () => {
      expect(wrapper).toMatchSnapshot();
   });

   test("Debe mostrar el valor por defecto 100 ", () => {
      const wrapper = shallow(<CounterApp value={100} />);

      const counterText = wrapper.find("h2").text().trim();
      expect(counterText).toBe("100");
   });

   test("Debe de incrementar con el boton de +1", () => {
      wrapper.find("button").at(0).simulate("click");
      const counterText = wrapper.find("h2").text().trim()  ;

      expect(counterText).toBe("11");
   });

   test("Debe de decrementar con el boton de -1", () => {
      wrapper.find("button").at(2).simulate("click");
      const counterText = wrapper.find("h2").text().trim();
    
      expect(counterText).toBe("9");
   });

   test('Debe resetear el valor del contador con el boton reset', () => {
      const wrapper = shallow(<CounterApp value={105} />);
      wrapper.find("button").at(0).simulate("click");
      wrapper.find("button").at(0).simulate("click");
     

      wrapper.find("button").at(1).simulate("click");
      const counterText = wrapper.find("h2").text().trim();
    
      expect(counterText).toBe('105')
   })
   
});
