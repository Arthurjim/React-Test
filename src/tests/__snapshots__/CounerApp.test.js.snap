// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <CounterApp /> Debe ser igual el Counter app 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
